packages/ui/src/components/va-date-input/hooks/range-model-value-guard.ts

```diff
@@ -13,8 +13,15 @@ export type useRangeModelValueGuardProps = {
 export const useRangeModelValueGuard = (
   modelValue: Ref<VaDatePickerModelValue | undefined>,
   disabled: Ref<boolean>,
+  inputDateParse?: ((input: string | undefined | VaDatePickerModelValue) => Ref<VaDatePickerModelValue>) | undefined,
 ) => {
-  const bufferValue = ref<VaDatePickerModelValue | undefined>(modelValue.value)
+  let bufferValue: Ref<VaDatePickerModelValue | undefined>
+
+  if (inputDateParse !== undefined) {
+    bufferValue = inputDateParse(modelValue.value)
+  } else {
+    bufferValue = ref<VaDatePickerModelValue | undefined>(modelValue.value)
+  }
```

Comment:
Probably we need to use the `computed` (instead of `ref`) here with getters and seters